pallets/msa/src/lib.rs

```diff
@@ -1125,18 +1113,17 @@ impl<T: Config> Pallet<T> {
 	// pub fn fetch_msa_keys(msa_id: MessageSourceId) -> Vec<KeyInfoResponse<T::AccountId>> {
 	// 	let mut response = Vec::new();
 	// 	for key in Self::get_msa_keys(msa_id) {
-	// 		if let Ok(_info) = Self::try_get_msa_from_public_key(&key) {
+	// 		if let Ok(_info) = Self::ensure_valid_msa_key(&key) {
 	// 			response.push(KeyInfoResponse { key, msa_id });
 	// 		}
 	// 	}
 
 	// 	response
 	// }
 
-	/// Checks that a key is associated to an MSA and has not been revoked.
+	/// Retrieve MSA Id associated with `key` or return `NoKeyExists`
 	pub fn ensure_valid_msa_key(key: &T::AccountId) -> Result<MessageSourceId, DispatchError> {
-		let msa_id = Self::try_get_msa_from_public_key(key)?;
-
+		let msa_id = Self::get_msa_by_public_key(key).ok_or(Error::<T>::NoKeyExists)?;
```

Comment:
Adding the new error here, it would be nice to bubble up the error listing for methods that call this one.

- `create_provider`
- `revoke_delegation_by_delegator`

Reply:
Moved to #613